client.go

```diff
@@ -0,0 +1,331 @@
+}
+
+// NewVisualizationClient returns client with token
+func NewVisualizationClient(url string, token AuthToken) (*VisualizationClient, error) {
+	client := http.Client{}
+	return &VisualizationClient{client: &client, url: url, token: token}, nil
+
+}
+
+// A JSONMessage contains the json error message received when http request failed
+type JSONMessage struct {
+	Message string `json:"message"`
+}
+
+func (v *VisualizationClient) reIssue() bool {
+	_, err := v.OpenstackAuth()
+	return err == nil
+}
+
+// authorizedHttpRequest for token checking and reauth
+func (v *VisualizationClient) authorizedHttpRequest() (err error) {
+	// validate token
+	if v.token == (AuthToken{}) {
+		return NotFound{}
+	}
+	tokenExpires := v.token.Token.ExpiresAt.UnixNano() / 1000000
+	now := time.Now().UnixNano() / 1000000
+	if tokenExpires < now {
+		if v.reIssue() {
+			return v.authorizedHttpRequest()
+		}
+	}
+	return err
+}
+
+// httpRequest handle the request to Grafana server.
+//It returns the response body and a error if something went wrong
+func (v *VisualizationClient) httpRequest(method string, url string, body io.Reader) (result io.Reader, err error) {
+	err = v.authorizedHttpRequest()
+	if err != nil {
+		return
+	}
+	request, err := http.NewRequest(method, url, body)
+	request.Header.Set("Content-Type", "application/json")
+	request.Header.Add("X-OpenStack-Auth-Token", v.openstackToken)
+	bearer := fmt.Sprintf("Bearer %v", v.token)
+	request.Header.Add("Authorization", bearer)
+
+	response, err := v.client.Do(request)
+	if err != nil {
+		return result, VisualizationError{"0", "Unable to perform the http request", "request failed"}
+	}
+
+	//	defer response.Body.Close()
+	if response.StatusCode != 200 {
+		dec := json.NewDecoder(response.Body)
+		var Mess JSONMessage
+		dec.Decode(&Mess)
+
+		return result, VisualizationError{strconv.Itoa(response.StatusCode), Mess.Message, "response not 200"}
+	}
+	result = response.Body
+	return
+}
+
+// AuthToken for requests
+type AuthToken struct {
+	JWT string `json:"jwt"`
+	Token
+}
+
+// Token for AuthToken
+type Token struct {
+	OrganizationID string    `json:"organizationId"`
+	ExpiresAt      time.Time `json:"expiresAt"`
+	IsAdmin        bool      `json:"isAdmin"`
+}
+
+// UserInOrganization Get Users in organization
+type UserInOrganization struct {
+	OrgID    string `json:"orgID"`
+	UserID   string `json:"userID"`
+	Login    string `json:"login"`
+	Role     string `json:"role"`
+	Email    string `json:"email"`
+	Password string `json:"password"`
+}
+
+// Org Get organization list
+type Org struct {
+	ID   string `json:"ID"`
+	Name string `json:"name"`
+}
+
+// User gets Users List
+type User struct {
+	ID       string `json:"ID"`
+	Email    string `json:"email"`
+	Name     string `json:"name"`
+	Login    string `json:"login"`
+	Password string `json:"password"`
+}
+
+// OpenstackAuth gets a openstack token
+func (v *VisualizationClient) OpenstackAuth() (token AuthToken, err error) {
```

Comment:
Please hide this method and introduce `func (v *VisualizationClient)Authenticate() (error)` instead